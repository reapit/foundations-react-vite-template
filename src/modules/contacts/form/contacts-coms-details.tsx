import {
  BodyText,
  InputWrap,
  InputGroup,
  Label,
  Select,
  FormLayout,
  elFadeIn,
  elMb11,
  InputError,
  Toggle,
  ElToggleItem,
} from '@reapit/elements'
import { FC } from 'react'
import { UseFormReturn } from 'react-hook-form'
import { ContactFormSchema } from './contacts-validation-schema'

export const MARKETING_OPTIONS = [
  { value: 'grant', name: 'Granted' },
  { value: 'deny', name: 'Denied' },
  { value: 'notAsked', name: 'Not Asked' },
]

interface ContactsComsDetailsProps {
  form: UseFormReturn<ContactFormSchema, any>
}

export const ContactsComsDetails: FC<ContactsComsDetailsProps> = ({ form }) => {
  const {
    register,
    formState: { errors },
  } = form
  return (
    <>
      <div className={elMb11}>
        <BodyText hasGreyText hasNoMargin>
          Provide here the communication details and marketing preferences of your contact.
        </BodyText>
      </div>
      <FormLayout className={elFadeIn}>
        <InputWrap>
          <InputGroup
            label="Home Phone"
            placeholder="Please enter a telephone number"
            {...register('homePhone')}
            icon={errors?.homePhone?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.homePhone?.message && <InputError message={errors.homePhone.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Work Phone"
            placeholder="Please enter a telephone number"
            {...register('workPhone')}
            icon={errors?.workPhone?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.workPhone?.message && <InputError message={errors.workPhone.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Mobile Phone"
            placeholder="Please enter a telephone number"
            {...register('mobilePhone')}
            icon={errors?.mobilePhone?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.mobilePhone?.message && <InputError message={errors.mobilePhone.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup
            label="Email"
            type="email"
            placeholder="Please enter an email address"
            {...register('email')}
            icon={errors?.email?.message ? 'asteriskSystem' : undefined}
            intent="danger"
          />
          {errors.email?.message && <InputError message={errors.email.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Select {...register('marketingConsent')}>
              <option key="default-option" value="">
                None selected
              </option>
              {MARKETING_OPTIONS.map(({ name, value }) => (
                <option key={value} value={value}>
                  {name}
                </option>
              ))}
            </Select>
            <Label>Marketing Consent</Label>
          </InputGroup>
          {errors.marketingConsent?.message && <InputError message={errors.marketingConsent.message} />}
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferenceLetter')} id="com-pref-letter">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference Letter</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferenceEmail')} id="com-pref-email">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference Email</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferencePhone')} id="com-pref-phone">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference Phone</Label>
          </InputGroup>
        </InputWrap>
        <InputWrap>
          <InputGroup>
            <Toggle {...register('communicationPreferenceSMS')} id="com-pref-sms">
              <ElToggleItem>On</ElToggleItem>
              <ElToggleItem>Off</ElToggleItem>
            </Toggle>
            <Label>Communicate Preference SMS</Label>
          </InputGroup>
        </InputWrap>
      </FormLayout>
    </>
  )
}
