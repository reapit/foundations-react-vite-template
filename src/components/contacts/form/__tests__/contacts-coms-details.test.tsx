import { ContactFormSchema } from '../contacts-validation-schema'
import { UseFormReturn } from 'react-hook-form'
import { ContactsComsDetails } from '../contacts-coms-details'
import { render } from '../../../../scripts/tests'

const error = {
  message: 'Error',
}

describe('ContactsComsDetails', () => {
  it('should match a snapshot', () => {
    const form = {
      register: jest.fn(() => ({})),
      formState: {
        errors: {
          email: error,
          mobilePhone: error,
          workPhone: error,
          homePhone: error,
          marketingConsent: error,
        },
      },
    } as unknown as UseFormReturn<ContactFormSchema, any>

    expect(render(<ContactsComsDetails form={form} />)).toMatchSnapshot()
  })
})
