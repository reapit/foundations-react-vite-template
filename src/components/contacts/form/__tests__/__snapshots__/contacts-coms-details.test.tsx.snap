// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsComsDetails should match a snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mocked-styled-97 el-snack-holder"
      />
      <p
        class="el-text-base el-has-bold-text el-has-section-margin el-body-text"
      >
        Provide here the communication details and marketing preferences of your contact.
      </p>
      <div
        class="el-fade-in el-form-layout"
      >
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="el-has-input-error el-input"
              id="test-static-id"
              placeholder="Please enter a telephone number"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Home Phone
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="el-has-input-error el-input"
              id="test-static-id"
              placeholder="Please enter a telephone number"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Work Phone
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="el-has-input-error el-input"
              id="test-static-id"
              placeholder="Please enter a telephone number"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Mobile Phone
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="el-has-input-error el-input"
              id="test-static-id"
              placeholder="Please enter an email address"
              type="email"
            />
            <label
              class="mocked-styled-41 el-label"
              for="test-static-id"
            >
              Email
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <select
              class="mocked-styled-87 el-select"
            >
              <option
                value=""
              >
                None selected
              </option>
              <option
                value="grant"
              >
                Granted
              </option>
              <option
                value="deny"
              >
                Denied
              </option>
              <option
                value="notAsked"
              >
                Not Asked
              </option>
            </select>
            <label
              class="mocked-styled-41 el-label"
            >
              Marketing Consent
            </label>
          </div>
          <p
            class="mocked-styled-42 el-input-error"
          >
            Error
          </p>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="mocked-styled-129 el-toggle-checkbox"
              id="com-pref-letter"
              type="checkbox"
            />
            <label
              class="mocked-styled-130 el-toggle-label"
              for="com-pref-letter"
            >
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                On
              </span>
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                Off
              </span>
            </label>
            <label
              class="mocked-styled-41 el-label"
            >
              Communicate Preference Letter
            </label>
          </div>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="mocked-styled-129 el-toggle-checkbox"
              id="com-pref-email"
              type="checkbox"
            />
            <label
              class="mocked-styled-130 el-toggle-label"
              for="com-pref-email"
            >
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                On
              </span>
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                Off
              </span>
            </label>
            <label
              class="mocked-styled-41 el-label"
            >
              Communicate Preference Email
            </label>
          </div>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="mocked-styled-129 el-toggle-checkbox"
              id="com-pref-phone"
              type="checkbox"
            />
            <label
              class="mocked-styled-130 el-toggle-label"
              for="com-pref-phone"
            >
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                On
              </span>
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                Off
              </span>
            </label>
            <label
              class="mocked-styled-41 el-label"
            >
              Communicate Preference Phone
            </label>
          </div>
        </div>
        <div
          class="mocked-styled-137 el-input-wrap"
        >
          <div
            class="mocked-styled-40 el-input-group"
          >
            <input
              class="mocked-styled-129 el-toggle-checkbox"
              id="com-pref-sms"
              type="checkbox"
            />
            <label
              class="mocked-styled-130 el-toggle-label"
              for="com-pref-sms"
            >
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                On
              </span>
              <span
                class="mocked-styled-127 el-toggle-item"
              >
                Off
              </span>
            </label>
            <label
              class="mocked-styled-41 el-label"
            >
              Communicate Preference SMS
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mocked-styled-97 el-snack-holder"
    />
    <p
      class="el-text-base el-has-bold-text el-has-section-margin el-body-text"
    >
      Provide here the communication details and marketing preferences of your contact.
    </p>
    <div
      class="el-fade-in el-form-layout"
    >
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="el-has-input-error el-input"
            id="test-static-id"
            placeholder="Please enter a telephone number"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Home Phone
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="el-has-input-error el-input"
            id="test-static-id"
            placeholder="Please enter a telephone number"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Work Phone
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="el-has-input-error el-input"
            id="test-static-id"
            placeholder="Please enter a telephone number"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Mobile Phone
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="el-has-input-error el-input"
            id="test-static-id"
            placeholder="Please enter an email address"
            type="email"
          />
          <label
            class="mocked-styled-41 el-label"
            for="test-static-id"
          >
            Email
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <select
            class="mocked-styled-87 el-select"
          >
            <option
              value=""
            >
              None selected
            </option>
            <option
              value="grant"
            >
              Granted
            </option>
            <option
              value="deny"
            >
              Denied
            </option>
            <option
              value="notAsked"
            >
              Not Asked
            </option>
          </select>
          <label
            class="mocked-styled-41 el-label"
          >
            Marketing Consent
          </label>
        </div>
        <p
          class="mocked-styled-42 el-input-error"
        >
          Error
        </p>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="mocked-styled-129 el-toggle-checkbox"
            id="com-pref-letter"
            type="checkbox"
          />
          <label
            class="mocked-styled-130 el-toggle-label"
            for="com-pref-letter"
          >
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              On
            </span>
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              Off
            </span>
          </label>
          <label
            class="mocked-styled-41 el-label"
          >
            Communicate Preference Letter
          </label>
        </div>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="mocked-styled-129 el-toggle-checkbox"
            id="com-pref-email"
            type="checkbox"
          />
          <label
            class="mocked-styled-130 el-toggle-label"
            for="com-pref-email"
          >
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              On
            </span>
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              Off
            </span>
          </label>
          <label
            class="mocked-styled-41 el-label"
          >
            Communicate Preference Email
          </label>
        </div>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="mocked-styled-129 el-toggle-checkbox"
            id="com-pref-phone"
            type="checkbox"
          />
          <label
            class="mocked-styled-130 el-toggle-label"
            for="com-pref-phone"
          >
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              On
            </span>
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              Off
            </span>
          </label>
          <label
            class="mocked-styled-41 el-label"
          >
            Communicate Preference Phone
          </label>
        </div>
      </div>
      <div
        class="mocked-styled-137 el-input-wrap"
      >
        <div
          class="mocked-styled-40 el-input-group"
        >
          <input
            class="mocked-styled-129 el-toggle-checkbox"
            id="com-pref-sms"
            type="checkbox"
          />
          <label
            class="mocked-styled-130 el-toggle-label"
            for="com-pref-sms"
          >
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              On
            </span>
            <span
              class="mocked-styled-127 el-toggle-item"
            >
              Off
            </span>
          </label>
          <label
            class="mocked-styled-41 el-label"
          >
            Communicate Preference SMS
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
